### ISSUE TITLE
RadzenDropDown: when using multi-select, collections bound to value are set to null on clear

### ISSUE THREAD

**Issue by pianomanjh** [radzenhq/radzen-blazor#1401]
**Describe the bug**
When using `RadzenDropDown` to make multiple selections bound to a collection, with `AllowClear` set to true, a user can click to clear all choices from the collection.  This results in the bound collection property being set to null, instead of clearing the collection entries.  I understand this may be by design, but, especially when using nullable annotations in a codebase, results in easy-to-miss null reference exceptions.

**To Reproduce**
```
public class Foo
{
    public ICollection<string> Settings { get; set; } = new List<string>();
}

var foo = new Foo() { Settings = ["Two"] };
var values = new string[] { "One", "Two", "Three" };

<RadzenDropDown Multiple="true" AllowClear="true" @bind-Value="@foo" Data="@values" />
```
After clicking the X in the control to clear the selections, `foo.Settings` is null instead of an empty collection.

**Expected behavior**
If the property bound to value started as a non-null collection instance, I would expect that collection to remain, but be empty.

I understand this may be breaking behavior, so, as a potential mitigation, if the property bound to value started as null, we could have that property be set to null on clear...

**Additional context**
If this is something the team is ok changing, let me know and I'm happy to post a PR.

**Comment by Riedi2070** [comment]
#vote up

**Comment by enchev** [comment]
Blazor bindings works only null/not null expression - they will not understand if a property is collection, if you clear the collection, remove or add items. We prefer not to do that since it will be exception compared to all other two-way bindings.

**Comment by pianomanjh** [comment]
> #vote up

@Riedi2070 this was resolved with #2261

**Comment by Riedi2070** [comment]
> > #vote up
> 
> [@Riedi2070](https://github.com/Riedi2070) this was resolved with [#2261](https://github.com/radzenhq/radzen-blazor/pull/2261)

Thx u very much
